scipy/stats/tests/test_stats.py

```diff
@@ -7758,6 +7758,80 @@ def test_alias(self):
         assert_equal(res.stat, res.statistic)
 
 
+class TestConfInt:
```

Comment:
@romain-jacob Here in the tests, can you:

- exercise all the error validation code (e.g. systematically like [this](https://github.com/scipy/scipy/blob/c1ed5ece8ffbf05356a22a8106affcd11bd3aee0/scipy/stats/tests/test_fit.py#L673))
- Adapt the existing test to the new interface
- Compare the confidence intervals against those generated by [`scipy.stats.bootstrap`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html) (with coarse tolerances)
- Compare the p-value to that of `binomtest` (non-edge cases, of course)
- Test against reference results from the literature (especially edge cases).

```suggestion
class TestQuantileTest:
```